.github/workflows/run-benchmarks.yml

```diff
@@ -0,0 +1,74 @@
+name: Run benchmarks
+# ToDo: add an arbitrary label to keep this cache separate from the builds
+on:
+  workflow_dispatch:
+  # Manually triggered by an authorized repo contributor.
+  push:
+    branches: main # To build the "benchmark-only" cache
+  pull_request:
+    paths:
+      - '.github/workflows/run-benchmarks.yml'
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.head.label || github.head_ref || github.ref }}
+  cancel-in-progress: true      
+jobs:
+  benchmark:
+    permissions:
+      pull-requests: write
+      contents: write
+      id-token: write
+    runs-on: ubuntu-latest-8-cores
+    strategy:
+      matrix:
+        runtimes: [centrifuge, altair]
+    env:       
+      # RUST_TOOLCHAIN: "1.66.0"
+      RUSTC_WRAPPER: "sccache"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac #v4
+
+      - name: Prep build on Ubuntu
+        uses: ./.github/actions/prep-ubuntu
+        with:
+          # RUST_TOOLCHAIN: ${{ env.RUST_TOOLCHAIN }}
+          cache: enabled
+          # Cache needs Google credentials:
+          GWIP: ${{ secrets.GWIP_SCCACHE }}
+          GSA: ${{ secrets.GSA_SCCACHE }}
+
+      # - name: Checkout code
+      #   uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac #v4
+      #   with:
+      #     ref: 'main'
```

Comment:
Can be removed or needs retoggling?

Reply:
can be removed, it was in case we wanted to force it to always publish the benchmarks from the main branch code